aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/RequestRetry.kt

```diff
@@ -0,0 +1,45 @@
+package com.amplifyframework.datastore.syncengine
+
+import com.amplifyframework.AmplifyException
+
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.core.SingleEmitter
+import java.util.concurrent.TimeUnit
+import kotlin.math.pow
+
+
+class RequestRetry(private val maxExponent: Int = 8,
+                    private val jitterFactor: Int = 100,
+                    private val maxAttempts: Int = 3) {
+
+    private var numberOfAttempts = 0
+
+
+        fun <T> retry(single: Single<T>, skipExceptions: List<Class<out Throwable>?>): Single<T> {
+            return Single.create { emitter -> call(single, emitter, 0, skipExceptions) }
+        }
+
+        private fun <T> call(single: Single<T>, emitter: SingleEmitter<T>, delayInSeconds: Long, skipExceptions: List<Class<out Throwable>?> ) {
+           single.delaySubscription(delayInSeconds, TimeUnit.SECONDS)
+                .subscribe({
+                            t: T -> emitter.onSuccess(t)
+                }) { error ->
+                    numberOfAttempts++
+                    if (numberOfAttempts > maxAttempts || (error.javaClass in skipExceptions)) {
+                        emitter.onError(error)
+                    } else {
+                        call(single, emitter, jitteredDelay(),skipExceptions )
+                    }
+                }
+        }
+
+
+    fun jitteredDelay(): Long {
+        val waitTimeSeconds: Long = java.lang.Double.valueOf(
```

Comment:
Can you import `java.lang.Double` at the top so that you don't have to use the fully qualified package name here?

Reply:
Done